src/mrinufft/trajectories/display3D.py

```diff
@@ -0,0 +1,122 @@
+    backend: str = "gpunufft",
+    traj_params: dict = None,
+    turbo_factor: int = 176,
+    elliptical_samp: bool = True,
+    threshold: float = 1e-3,
+):
+    """
+    Compute the gridded trajectory for MRI reconstruction.
+
+    Parameters
+    ----------
+    shots : ndarray
+        The input array of shape (N, M), where N is the number of shots and M is the
+        number of samples per shot.
+    shape : tuple
+        The desired shape of the gridded trajectory.
+    grid_type : str, optional
+        The type of gridded trajectory to compute. Default is "density".
+        It can be one of the following:
+            "density" : Get the sampling density in closest number of samples per voxel.
+            Helps understand suboptimal sampling.
+            "time" : Get the sampling in time, this is helpful to view and understand
+            off-resonance effects.
+            "inversion" : Relative inversion time at the sampling location. Needs
+            turbo_factor to be set.
+            "holes": Show the k-space holes within a elliosoid of the k-space.
+            "gradients": Show the gradient strengths of the k-space trajectory.
+            "slew": Show the slew rate of the k-space trajectory.
+    osf : int, optional
+        The oversampling factor for the gridded trajectory. Default is 1.
+    backend : str, optional
+        The backend to use for gridding. Default is "gpunufft".
+        Note that "gpunufft" is anyway used to get the `pipe` density internally.
+    traj_params : dict, optional
+        The trajectory parameters. Default is None.
+        This is only needed when `grid_type` is "gradients" or "slew".
+    turbo_factor : int, optional
+        The turbo factor when sampling is with inversion. Default is 176.
+    elliptical_samp : bool, optional
+        Whether to use elliptical sampling. Default is True.
+        This is useful while analyzing the k-space holes.
+    threshold: float, optional
+        The threshold for the k-space holes. Default is 1e-3.
+
+    Returns
+    -------
+    ndarray
+        The gridded trajectory of shape `shape`.
+    """
+    samples = shots.reshape(-1, shots.shape[-1])
+    dcomp = get_density("pipe")(samples, shape)
+    grid_op = get_operator(backend)(
+        samples, [sh * osf for sh in shape], density=dcomp, upsampfac=1
+    )
+    gridded_ones = grid_op.raw_op.adj_op(np.ones(samples.shape[0]), None, True)
+    if grid_type == "density":
+        return np.abs(gridded_ones).squeeze()
+    elif grid_type == "time":
+        data = grid_op.raw_op.adj_op(
+            np.tile(np.linspace(1, 10, shots.shape[1]), (shots.shape[0],)),
+            None,
+            True,
+        )
+    elif grid_type == "inversion":
+        data = grid_op.raw_op.adj_op(
+            np.repeat(
+                np.linspace(1, 10, turbo_factor), samples.shape[0] // turbo_factor + 1
+            )[: samples.shape[0]],
+            None,
+            True,
+        )
```

Comment:
Same problem here, the output shape doesn't seem to match the documented output

Reply:
Same as above, isnt data.shape same as shape we give as input?